src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -2515,7 +2515,8 @@ public void addPreviewInfo(Element forWhat, Content target) {
             Content note2 = contents.getContent("doclet.PreviewTrailingNote2", nameCode);
             previewDiv.add(HtmlTree.DIV(HtmlStyles.previewComment, note2));
             target.add(previewDiv);
-        } else if (forWhat.getKind().isClass() || forWhat.getKind().isInterface()) {
+        } else if ((forWhat.getKind().isClass() || forWhat.getKind().isInterface())
+                && !utils.nonPreviewExtendsPreview(forWhat)) {
```

Comment:
I think I fixed it in [f8b3110](https://github.com/openjdk/jdk/pull/22126/commits/f8b3110c389d65c6cddbafbb28f554aeb93ebab1). The solution looks simpler without any of the added methods, thanks.

> (An interface extending a preview interface is a different case again, although I don't think that should ever occur in the JDK.)

Is this a typo? because I think this is exactly what the [PEM Encodings JEP](https://github.com/openjdk/jdk/pull/17543/files) does.

```
public non-sealed interface AsymmetricKey extends Key, DEREncodable {
}
```